docs/dex-specification.md

```diff
@@ -846,12 +875,14 @@ function foo() public {
 
 Roles can be granted and revoked dynamically via the `grantRole` and `revokeRole` functions. Each role has an associated admin role, and only accounts that have the admin role can call `grantRole` and `revokeRole`.
 
-##### Admin role
+###### Admin role
 
 By default, the admin role for all roles is `DEFAULT_ADMIN_ROLE`, which means that only accounts with this role will be able to grant or revoke other roles. More complex role relationships can be created by using `_setRoleAdmin`.
 
 The `DEFAULT_ADMIN_ROLE` is also its own admin: it has permission to grant and revoke this role.
 
+**Warning**: With `revokeRole`, the admin can revoke their own role. If this happens, DEX operator losses control over DEX.
```

Comment:
With `revokeRole`, the admin can revoke their own role (any granted role). This leads to all associated functionality associated with the said role being inaccessible. For example, revoking or renouncing MINTER_ROLE makes all associated minting functionality inaccessible. 
Renouncing or revoking DEFAULT_ADMIN_ROLE role is critical, so the admin will no longer be able to grant or revoke any other roles.